mock_ros2_project/service_client.py

```
@@ -27,35 +34,81 @@ def __init__(self):
         
         self.req = AddTwoInts.Request()
 
-    def send_request(self, a, b):
-        """Send a request to the service."""
+    def send_request(self, a: int, b: int) -> Optional[AddTwoInts.Response]:
+        """Send a request to the service.
+        
+        Args:
+            a: First integer to add
+            b: Second integer to add
+            
+        Returns:
+            Optional[AddTwoInts.Response]: Service response or None if failed
+        """
         self.req.a = a
         self.req.b = b
         
-        self.future = self.cli.call_async(self.req)
-        rclpy.spin_until_future_complete(self, self.future)
+        self.get_logger().info(f'Sending request: {a} + {b}')
         
-        return self.future.result()
+        try:
+            self.future = self.cli.call_async(self.req)
+            rclpy.spin_until_future_complete(self, self.future)
+            
+            if self.future.result() is not None:
+                return self.future.result()
+            else:
+                self.get_logger().error('Service call failed')
+                return None
+        except Exception as e:
```

Catching the broad `Exception` class swallows all possible errors, including ROS-specific ones that might warrant special handling. Narrow the except clause to expected exceptions (e.g., `rclpy.service.ServiceError`) or re-raise unknown errors after logging to simplify debugging.